diff --git a/src/widget-theme.c b/src/widget-theme.c
index 3331ae4..bceadfe 100644
--- a/src/widget-theme.c
+++ b/src/widget-theme.c
@@ -36,7 +36,7 @@ static GSList* load_themes_in_dir(const char* theme_dir, GSList* themes)
             if(!g_slist_find_custom(themes, name, (GCompareFunc)strcmp))
             {
                 /* test if this is a gtk theme */
-                char* gtkrc = g_build_filename(theme_dir, name, "gtk-2.0/gtkrc", NULL);
+                char* gtkrc = g_build_filename(theme_dir, name, "gtk-3.0/gtk.css", NULL);
                 if(g_file_test(gtkrc, G_FILE_TEST_EXISTS))
                     themes = g_slist_prepend(themes, g_strdup(name));
                 g_free(gtkrc);
