diff --git a/hook-functions b/hook-functions
index 95bd584..1909e8d 100644
--- a/hook-functions
+++ b/hook-functions
@@ -200,7 +200,7 @@ copy_file() {
 
 			# Create a relative link so it always points
 			# to the right place
-			ln -rs "${DESTDIR}/${link_target}" "${DESTDIR}/${target}"
+			ln -s "${DESTDIR}/${link_target}" "${DESTDIR}/${target}"
 		fi
 
 		# Copy the link target if it doesn't already exist
@@ -863,7 +863,7 @@ cache_run_scripts()
 	[ ! -d "${initdir}" ] && return
 
 	true > "${initdir}/ORDER"
-	runlist=$(get_prereq_pairs | tsort)
+	runlist=$(get_prereq_pairs | sort)
 	for crs_x in ${runlist}; do
 		[ -f "${initdir}/${crs_x}" ] || continue
 		echo "${scriptdir}/${crs_x} \"\$@\"" >> "${initdir}/ORDER"
@@ -901,6 +901,6 @@ run_scripts()
 	initdir=${1}
 	[ ! -d "${initdir}" ] && return
 
-	runlist=$(get_prereq_pairs | tsort)
+	runlist=$(get_prereq_pairs | sort)
 	call_scripts "$scriptdir"
 }
