diff --git a/CMakeLists.txt b/CMakeLists.txt
index 811f181..d55a1f0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,7 +50,6 @@ find_package(KF5 ${KF5_MIN_VERSION} REQUIRED
     Plasma
     Service
     Solid
-    Wallet
     WidgetsAddons
     WindowSystem
     KCMUtils
diff --git a/kded/CMakeLists.txt b/kded/CMakeLists.txt
index f7dd217..69b739e 100644
--- a/kded/CMakeLists.txt
+++ b/kded/CMakeLists.txt
@@ -8,14 +8,12 @@ target_sources(kded_networkmanagement PRIVATE
     notification.cpp
     monitor.cpp
     passworddialog.cpp
-    secretagent.cpp
     service.cpp
     bluetoothmonitor.h
     connectivitymonitor.h
     notification.h
     monitor.h
     passworddialog.h
-    secretagent.h
     service.h
     modemmonitor.cpp
     modemmonitor.h
@@ -42,7 +40,6 @@ target_link_libraries(kded_networkmanagement
     KF5::Notifications
     KF5::Service
     KF5::Solid
-    KF5::Wallet
     KF5::WindowSystem
     KF5::KIOGui
     KF5::ModemManagerQt
diff --git a/kded/secretagent.cpp b/kded/secretagent.cpp
index 8c6b394..c41186f 100644
--- a/kded/secretagent.cpp
+++ b/kded/secretagent.cpp
@@ -657,4 +657,4 @@ void SecretAgent::importSecretsFromPlainTextFiles()
             group.deleteGroup();
         }
     }
-}
+}
\ No newline at end of file
diff --git a/kded/secretagent.h b/kded/secretagent.h
index 0d83af2..297fe6d 100644
--- a/kded/secretagent.h
+++ b/kded/secretagent.h
@@ -113,4 +113,4 @@ private:
     void importSecretsFromPlainTextFiles();
 };
 
-#endif // PLASMA_NM_SECRET_AGENT_H
+#endif // PLASMA_NM_SECRET_AGENT_H
\ No newline at end of file
diff --git a/kded/service.cpp b/kded/service.cpp
index 5c0db26..b57d784 100644
--- a/kded/service.cpp
+++ b/kded/service.cpp
@@ -14,7 +14,6 @@
 #include "connectivitymonitor.h"
 #include "monitor.h"
 #include "notification.h"
-#include "secretagent.h"
 
 #include <QDBusConnection>
 #include <QDBusConnectionInterface>
@@ -27,7 +26,6 @@ K_PLUGIN_CLASS_WITH_JSON(NetworkManagementService, "networkmanagement.json")
 class NetworkManagementServicePrivate
 {
 public:
-    SecretAgent *agent = nullptr;
     Notification *notification = nullptr;
     Monitor *monitor = nullptr;
     ConnectivityMonitor *connectivityMonitor = nullptr;
@@ -41,8 +39,6 @@ NetworkManagementService::NetworkManagementService(QObject *parent, const QVaria
 
     connect(this, &KDEDModule::moduleRegistered, this, &NetworkManagementService::init);
 
-    d->agent = new SecretAgent(this);
-    connect(d->agent, &SecretAgent::secretsError, this, &NetworkManagementService::secretsError);
 }
 
 NetworkManagementService::~NetworkManagementService()
diff --git a/libs/CMakeLists.txt b/libs/CMakeLists.txt
index 59d8ced..166a314 100644
--- a/libs/CMakeLists.txt
+++ b/libs/CMakeLists.txt
@@ -39,7 +39,6 @@ PRIVATE
     KF5::I18n
     KF5::Notifications
     KF5::Service
-    KF5::Wallet
     KF5::WindowSystem
     KF5::KIOGui
 )
diff --git a/libs/editor/CMakeLists.txt b/libs/editor/CMakeLists.txt
index 1e2c3d6..9d71ad8 100644
--- a/libs/editor/CMakeLists.txt
+++ b/libs/editor/CMakeLists.txt
@@ -150,7 +150,6 @@ PRIVATE
     KF5::KIOWidgets
     KF5::Notifications
     KF5::Solid
-    KF5::Wallet
     Qt::DBus
     Qt::Network
     qca-qt${QT_MAJOR_VERSION}
diff --git a/libs/editor/uiutils.cpp b/libs/editor/uiutils.cpp
index 95d5dec..ba99295 100644
--- a/libs/editor/uiutils.cpp
+++ b/libs/editor/uiutils.cpp
@@ -16,7 +16,6 @@
 #include <KLocalizedString>
 #include <KSharedConfig>
 #include <KUser>
-#include <KWallet>
 
 #include <NetworkManagerQt/Manager>
 #include <NetworkManagerQt/Security8021xSetting>
@@ -720,7 +719,7 @@ void UiUtils::setConnectionDefaultPermissions(NetworkManager::ConnectionSettings
         return;
     }
 
-    if (Configuration::self().systemConnectionsByDefault() || !KWallet::Wallet::isEnabled() || isLiveImage()) {
+    if (true || Configuration::self().systemConnectionsByDefault() || isLiveImage()) {
         auto modifySystem = NetworkManager::permissions().value(QStringLiteral("org.freedesktop.NetworkManager.settings.modify.system"));
         if (modifySystem == QLatin1String("yes")) {
             wifiSecurity->setLeapPasswordFlags(NetworkManager::Setting::SecretFlagType::None);
diff --git a/libs/editor/widgets/passwordfield.cpp b/libs/editor/widgets/passwordfield.cpp
index 7b8c98d..97db7ff 100644
--- a/libs/editor/widgets/passwordfield.cpp
+++ b/libs/editor/widgets/passwordfield.cpp
@@ -11,7 +11,6 @@
 
 #include <KAuthorized>
 #include <KLocalizedString>
-#include <KWallet>
 
 PasswordField::PasswordField(QWidget *parent, Qt::WindowFlags f)
     : QWidget(parent, f)
@@ -46,7 +45,7 @@ PasswordField::PasswordField(QWidget *parent, Qt::WindowFlags f)
     // Do not add by default
     // m_passwordOptionsMenu->addItem(QIcon::fromTheme(QStringLiteral("document-close")), i18n("This password is not required"), NotRequired);
 
-    if (KWallet::Wallet::isEnabled()) {
+    if (0) {
         m_passwordOptionsMenu->setCurrentIndex(0);
     } else {
         m_passwordOptionsMenu->setCurrentIndex(1);
diff --git a/libs/handler.cpp b/libs/handler.cpp
index 8b1782d..04be544 100644
--- a/libs/handler.cpp
+++ b/libs/handler.cpp
@@ -39,7 +39,6 @@
 #include <KPluginMetaData>
 #include <KProcess>
 #include <KUser>
-#include <KWallet>
 #include <KWindowSystem>
 
 #include <nm-client.h>
